nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/encryption/EncryptedS3PutEnrichmentService.java

```diff
@@ -0,0 +1,181 @@
+            .displayName("Encryption Method")
+            .required(true)
+            .allowableValues(METHOD_SSE_S3, METHOD_SSE_KMS, METHOD_SSE_C)
+            .defaultValue(METHOD_SSE_S3)
+            .description("Method by which the S3 object will be encrypted server-side.")
+            .build();
+
+    public static final PropertyDescriptor ALGORITHM = new PropertyDescriptor.Builder()
+            .name("sse-algorithm")
+            .displayName("Algorithm")
+            .allowableValues(ALGORITHM_AES256)
+            .defaultValue(ALGORITHM_AES256)
+            .description("Encryption algorithm to use (only AES256 currently supported)")
+            .build();
+
+    public static final PropertyDescriptor KMS_KEY_ID = new PropertyDescriptor.Builder()
+            .name("sse-kme-key-id")
+            .displayName("KMS Key Id")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("Custom KMS key identifier. Supports key or alias ARN.")
+            .build();
+
+    public static final PropertyDescriptor CUSTOMER_KEY = new PropertyDescriptor.Builder()
+            .name("sse-customer-key")
+            .displayName("Customer Key")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .description("Customer provided 256-bit, base64-encoded encryption key.")
+            .build();
+
+    public static final PropertyDescriptor CUSTOMER_ALGORITHM = new PropertyDescriptor.Builder()
+            .name("sse-customer-algorithm")
+            .displayName("Customer Algorithm")
+            .allowableValues(CUSTOMER_ALGORITHM_AES256)
+            .defaultValue(CUSTOMER_ALGORITHM_AES256)
+            .description("Customer encryption algorithm to use (only AES256 currently supported)")
+            .build();
+
+    private static final List<PropertyDescriptor> properties;
+    private String encryptionMethod;
+    private String algorithm;
+    private String kmsKeyId;
+    private String customerKey;
+    private String customerAlgorithm;
+
+
+    static {
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(ENCRYPTION_METHOD);
+        props.add(ALGORITHM);
+        props.add(KMS_KEY_ID);
+        props.add(CUSTOMER_KEY);
+        props.add(CUSTOMER_ALGORITHM);
+        properties = Collections.unmodifiableList(props);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @OnEnabled
+    public void onConfigured(final ConfigurationContext context) throws InitializationException {
+        encryptionMethod = context.getProperty(ENCRYPTION_METHOD).getValue();
+        algorithm = context.getProperty(ALGORITHM).getValue();
+        kmsKeyId = context.getProperty(KMS_KEY_ID).getValue();
```

Comment:
Should this evaluate expression language like `context.getProperty(KMS_KEY_ID).evaluateAttributeExpressions().getValue()`?